### PR TITLE
add: latest casts list and small styling improvements

### PR DESCRIPTION
**Pull request by dylsteck**
adds: 
- auto-refreshing list of latest casts on the homepage(linking to the result page in the explorer)
- small styling adjustments such as a new hero card for the search bar on the home page and padding around certain objects
Screenshot 2024-06-04 at 5 21 21 PM

### CODE REVIEW

File: src/app/layout.tsx [hunk 1]
```diff
@@ -13,37 +14,35 @@ export default function RootLayout({
...
+      <body className="relative h-screen flex flex-col items-center min-h-screen">
         <Providers>
           <div className="absolute top-0 left-0 p-2">
             <Link href="https://www.neynar.com" target="_blank">
-              <img className="w-16 md:w-32" src={"/neynar.png"} />
+              <img className="w-16 md:w-32" src="/neynar.png" />
             </Link>
           </div>
-          <div className="absolute top-5 left-1/2 transform -translate-x-1/2 p-2">
+          <Card className="min-w-[55%] md:min-w-[40%] fixed top-5 left-1/2 transform -translate-x-1/2 p-4 mb-5 z-10">
```
Comment (kevoconnell) [inline src/app/layout.tsx]: can you make this absolute instead of fixed

File: src/components/cast-component.tsx [hunk 2]
```diff
@@ -1,32 +1,34 @@
 import { AvatarImage, AvatarFallback, Avatar } from "@/components/ui/avatar"
-import { CardHeader, CardContent, CardFooter, Card } from "@/components/ui/card"
-//note: not using this in favor of https://github.com/pugson/react-farcaster-embed, but this is a good example of how to render a cast.
-import Link from 'next/link';
+import { CardHeader, CardContent, Card } from "@/components/ui/card"
+import Link from 'next/link'
```
Comment (kevoconnell) [inline src/components/cast-component.tsx:4]: nice

File: src/components/latest-casts.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
+/* eslint-disable @next/next/no-img-element */
+import useSWR from 'swr';
+import axios from 'axios';
+import { useEffect, useState } from 'react';
+import { CastComponent } from './cast-component';
```
Comment (kevoconnell) [inline src/components/latest-casts.tsx:5]: nice

File: src/app/layout.tsx (hunk 1 above)
Comment (kevoconnell) [reply src/app/layout.tsx]: expanding on this I know this wasn't good on mobile before 😬 , can you try to improve it?

**Review by kevoconnell — CHANGES_REQUESTED** [review]
looks really good small thing , can you also take a screenshot of what it would look like on mobile?

File: src/app/layout.tsx (hunk 1 above)
Comment (dylsteck) [reply src/app/layout.tsx]: yeah for sure, can make those changes and work on the mobile restyling/add a screenshot after :D

**Comment by dylsteck** [comment]
Screenshot 2024-06-05 at 4 07 17 PM
Screenshot 2024-06-05 at 4 07 25 PM

Screenshot 2024-06-05 at 4 07 36 PM

made some new mobile responsiveness / content alignment changes and am adding screenshots of them here cc @MrKevinOConnell

**Review by kevoconnell — APPROVED** [review]
lgtm!

File: src/app/layout.tsx [hunk 4]
```diff
@@ -13,37 +14,35 @@ export default function RootLayout({
...
         <meta property="og:title" content="Farcaster Explorer" />
-        <meta property="og:description" content="Explore event propogation on farcaster." />
+        <meta property="og:description" content="Explore event propagation on farcaster." />
         <meta property="og:image" content="/neynar.png" />
         <meta property="og:url" content="https://www.neynar.com" />
         <meta property="og:type" content="website" />
         <meta name="twitter:card" content="summary_large_image" />
         <meta name="twitter:title" content="Farcaster Explorer" />
-        <meta name="twitter:description" content="Explore event propogation on farcaster." />
+        <meta name="twitter:description" content="Explore event propagation on farcaster." />
```
Comment (kevoconnell) [inline src/app/layout.tsx:25]: nice

File: src/app/layout.tsx [hunk 5]
```diff
@@ -13,37 +14,35 @@ export default function RootLayout({
...
+              <Link href="https://www.neynar.com" target="_blank">
+                  <img className="w-16 md:w-32" src="/neynar.png" />
+              </Link>
+            </div>
+            <Card className="w-[100%] p-3">
+              <Link href="/" className="text-black font-bold">
+                <p className="mb-4 text-lg md:text-">Farcaster Explorer</p>
+              </Link>
+              <Search />
+            </Card>
```
Comment (kevoconnell) [inline src/app/layout.tsx:41]: can you change this to what it was before?

on the top left, the neynar logo should show, in the center is farcaster explorer

File: src/components/search.tsx [hunk 6]
```diff
@@ -33,15 +33,15 @@ export default function Search({
   return (
```
Comment (kevoconnell) [inline src/components/search.tsx:33]: this is nice

**Review by kevoconnell — COMMENTED** [review]
can you seperate the farcaster explorer and the search like it was before? other than that this looks good

Screenshot 2024-06-05 at 10 30 26 PM

File: src/app/page.tsx [hunk 7]
```diff
@@ -1,39 +1,29 @@
...
+          <CastComponent cast={exampleCast} headerText="Example hash" />
+          <CastComponent cast={warpcastURLCast} warpcastUrl={warpcastURLCastURL} headerText="Example Warpcast cast url" />
         </div>
+        <div className="flex flex-col gap-2 md:flex-row md:gap-0 justify-center md:justify-between w-[100%]">
+          <ProfileComponent pfp={FIDPFP} url="3" headerText="Example FID" />
+          <ProfileComponent pfp={warpcastURLPFP} url={warpcastURLProfile} headerText="Example Warpcast profile url" />
         </div>
-      
-      </div>
+        <LatestCasts />
```
Comment (kevoconnell) [inline src/app/page.tsx:26]: can you make it so the latest casts are in the center of the screen, seems like they aren't rn 
Screenshot 2024-06-05 at 10 32 14 PM

File: src/app/[identifier]/page.tsx [hunk 8]
```diff
@@ -85,8 +85,8 @@ export default function Page({ params }: ResponseProps) {
   return (
     <>
       <Modal isOpen={isModalOpen} toggleModal={closeModal} response={modalData} title={modalTitle} />
-      <div className="flex flex-col w-full items-center justify-center">
-        <div className="w-full flex md:flex-row flex-col justify-center items-center md:space-x-0 ">
+      <div className="flex flex-col w-full items-center justify-center mt-[25%] md:mt-[12.5%] lg:mt-[15%]">
+        <div className="w-full flex md:flex-row gap-4 flex-col justify-center items-center md:space-x-0 ">
```
Comment (kevoconnell) [inline src/app/[identifier]/page.tsx:89]: is the gap for mobile responsiveness? would prefer to remove this (nit though, just want it to be as similar to original designs)

**Pull request merged**